### PR TITLE
Update Doxygen file blocks to remove copyright and license information

### PR DESCRIPTION
**Pull request by dgreen-arm**
This PR separates the copyright and license information from the Doxygen documentation blocks. The documentation was also updated to have file and brief blocks for all files.

### CODE REVIEW

File: scripts/data_files/error.fmt [hunk 1]
```diff
@@ -1,6 +1,9 @@
-/*
- *  Error message information
+/**
+ * \file error.c
```
[the remark this answers is not in the document]
Comment (dgreen-arm) [reply scripts/data_files/error.fmt]: This file is used by the generate_errors.pl script to create the error.c file, hence the filename being error.c. The generate_features.pl script follows a similar process with version_features.fmt and version_features.c.

[the remark this answers is not in the document]
Comment (Patater) [reply scripts/data_files/error.fmt]: OK

File: ChangeLog [hunk 2]
```diff
@@ -78,6 +78,7 @@ Changes
      and the message digest. Further, allow enabling/disabling of authority
      identifier, subject identifier and basic constraints extensions.
    * Only run AES-192 self-test if AES-192 is available. Fixes #963.
+   * Remove copyright and license information from Doxygen documentation.
```
Comment (Patater) [inline ChangeLog]: This change doesn't warrant a ChangeLog entry.

**Comment by Patater** [comment]
Closed in favor of separate PR.

**Pull request closed without merging**